subsys/net/lib/lwm2m/lwm2m_message_handling.c

```diff
@@ -2549,13 +2652,19 @@ int generate_notify_message(struct lwm2m_ctx *ctx, struct observe_node *obs, voi
 	}
 
 	if (ret < 0) {
+		if (lwm2m_timeseries_data_rebuild(msg, ret)) {
+			goto msg_init;
+		}
```

Comment:
I think some comment explaining logic here could be useful, it's not clear to me what is the retry condition.

Reply:
Ok I will add comment's for this.